datalad/distribution/create.py

```diff
@@ -0,0 +1,86 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""High-level interface for dataset creation
+
+"""
+
+__docformat__ = 'restructuredtext'
+
+import logging
+import os
+from datalad.distribution.dataset import Dataset, datasetmethod, EnsureDataset
+from datalad.interface.base import Interface
+from datalad.support.constraints import EnsureStr, EnsureNone, EnsureDType
+from datalad.support.param import Parameter
+from datalad.support.annexrepo import AnnexRepo
+from datalad.support.gitrepo import GitRepo
+
+lgr = logging.getLogger('datalad.distribution.create')
+
+
+class Create(Interface):
+    """Create a new dataset.
+
+    This command initializes a new repository at a given location, or the
+    current directory.
+    """
+
+    _params_ = dict(
+        loc=Parameter(
+            args=("loc",),
+            doc="""location where the dataset shall be  created. If `None`,
+            is given a dataset will be created in the current working
+            directory.""",
+            nargs='?',
+            # put dataset 2nd to avoid useless conversion
+            constraints=EnsureStr() | EnsureDataset() | EnsureNone()),
+        description=Parameter(
+            args=("-d", "--description",),
+            doc="""Short description that humans can use to identify the
+            repository/location, e.g. "Precious data on my laptop.""",
+            nargs=1),
+        no_annex=Parameter(
+            args=("--no-annex",),
+            doc="""Flag that if given a plain Git repository will be created
+            without any annex.""",
+            action='store_false'),
+        annex_version=Parameter(
+            args=("--annex-version",),
+            doc="""Select particular annex repository version. The list of
+            supported versions depends on the available git-annex version.""",
+            nargs=1,
+            constraints=EnsureDType(int) | EnsureNone()),
+        annex_backend=Parameter(
+            args=("--annex-backend",),
+            # not listing choices here on purpose to avoid future bugs
+            doc="""Set default hashing backend used by the new dataset.
+            For a list of supported backends see the git-annex
+            documentation.""",
+            nargs=1),
```

Comment:
shouldn't there be a '--register-within-superdataset' option?  may be even allowing for "auto(default),yes,no" whenever default would be to register within superdataset if it is under any (according to filesystem hierarchy).  If 'yes' -- then, would fail if there is no super dataset found.  Could even have/support 'ask'.  E.g. we could use ui.question if in interactive mode to ask if to register or not... ?